eth/gaspricemonitor.go

```diff
@@ -137,5 +138,10 @@ func (gpm *GasPriceMonitor) updateGasPrice(gasPrice *big.Int) {
 	gpm.gasPriceMu.Lock()
 	defer gpm.gasPriceMu.Unlock()
 
+	// Don't update the gas price if it is nil
+	if gasPrice == nil {
```

Comment:
The `SuggestGasPrice()` [impl](https://github.com/ethereum/go-ethereum/blob/04dcc9378dc38d54eeb6a688d81e8b5e13446519/ethclient/ethclient.go#L497) will not return a nil value when the error is nil so the error check on L124 should be sufficient.